src/service/layers-impl.js

```diff
@@ -788,7 +939,50 @@ export class LayoutElement {
       }
     }
 
-    layer.remeasure_();
+    if (layer.needsRemeasure_) {
+      layer.remeasure_();
+    }
+  }
+
+  /**
+   * Iterates the layout's parent layer ancestry, starting from the root down
+   * to the layout.
+   *
+   * This sets a whether the layer isActive during that layer's iteration. Any
+   * attempts to access #isActive outside of the iterator call will fail.
+   *
+   * @param {function(T, !LayoutElement, number, !AncestryStateDef):T} iterator
+   * @param {!AncestryStateDef} state
+   * @return {T}
+   * @template T
+   */
+  ancestry(iterator, state) {
+    const activeLayer = Services.layersForDoc(this.element_).getActiveLayer();
+
+    // Gather, and update whether the layers are descendants of the active
+    // layer.
+    let isActive = activeLayer === this || activeLayer.contains(this);
+
+    let layer = this;
+    while (layer) {
+      ANCESTRY_STORE.push(layer);
+
+      layer.isActive_ = isActive;
+      if (layer === activeLayer) {
+        isActive = false;
+      }
+
+      layer = layer.getParentLayer();
+    }
+
+    let value;
```

Comment:
`accumulator` since this is reduce-like, and explicitly initialize to `undefined`.